client/src/js/services/StockService.js

```diff
@@ -241,7 +248,7 @@ function StockService(Api, Filters, AppCache, Periods, $httpParamSerializer, Lan
 
   /** Get label for purchase Status */
   function statusLabelMap(status) {
-    var keys = {
+    const keys = {
```

Comment:
Since these are constants, can we move them outside of this function?  I feel like you structured your test similar to this code, but the code isn't well structured in this case.